qiskit/circuit/library/grover_operator.py

```diff
@@ -292,7 +295,7 @@ def _zero_reflection(
 
     num_ancillas = MCXGate.get_num_ancilla_qubits(len(qubits) - 1, mcx_mode)
     if num_ancillas > 0:
-        qr_ancilla = AncillaRegister(num_ancillas, "ancilla")
+        qr_ancilla: AncillaRegister | list[Bit] = AncillaRegister(num_ancillas, "ancilla")
```

Comment:
We could use an empty `AncillaRegister` in the else clause here to avoid the explicit type hint here